### PR TITLE
Add Dremio driver

### PR DESCRIPTION
**Pull request by Rodrigonavarro23**
## What type of PR is this? (check all applicable)

- [X] New Query Runner (Data Source)

## Description
Add [Dremio](https://www.dremio.com) as datasource. 
> "Dremio delivers lightning-fast queries and a self-service semantic layer directly on your data lake storage. No moving data to proprietary data warehouses, no cubes, no aggregation tables or extracts. Just flexibility and control for data architects, and self-service for data consumers."

In order to be able to use Dremio, I had to import pyodbc into the requirements and alien as dependency on the docker file, it is still necessary to install the driver on the machine, here I leave an example of installation in docker with Ubuntu image:

```
# install Dremio driver
ENV DRIVER_VERSION=1.4.2.1003
RUN wget "https://download.dremio.com/odbc-driver/${DRIVER_VERSION}/dremio-odbc-${DRIVER_VERSION}-1.x86_64.rpm" -O /dremio-odbc-${DRIVER_VERSION}-1.x86_64.rpm
RUN alien -i --scripts /dremio-odbc-${DRIVER_VERSION}-1.x86_64.rpm
RUN rm -f /dremio-odbc-${DRIVER_VERSION}-1.x86_64.rpm
```

## Mobile & Desktop Screenshots/Recordings (if there are UI changes)
Captura de Pantalla 2019-10-28 a la(s) 10 26 47
Captura de Pantalla 2019-10-28 a la(s) 10 26 55

### CODE REVIEW

File: redash/query_runner/dremio.py [hunk 1]
```diff
@@ -0,0 +1,119 @@
+from redash.query_runner import BaseQueryRunner, register
+from redash.query_runner import TYPE_STRING, TYPE_DATE, TYPE_DATETIME, TYPE_INTEGER, TYPE_FLOAT, TYPE_BOOLEAN
+from redash.utils import json_dumps, json_loads
+import pyodbc
+
+enabled = True
```
Comment (arikfr) [inline redash/query_runner/dremio.py]: You need to import this in a try/except block and set `enabled` accordingly. You can see examples for this pattern in other query runners.

File: redash/query_runner/dremio.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    noop_query = "SELECT 1"
+
+    @classmethod
+    def configuration_schema(cls):
+        return {
+            "type": "object",
+            "properties": {
+                "driver": {
+                    "type": "string",
+                    "default": "{Dremio ODBC Driver 64-bit}"
```
Comment (arikfr) [inline redash/query_runner/dremio.py]: Any reason someone might want to change this?

Comment (Rodrigonavarro23) [reply redash/query_runner/dremio.py]: Depend on the operating system, in osx is different from linux:
- Dremio ODBC Driver 64-bit
- Dremio ODBC Driver 32-bit
- Dremio ODBC Driver

**Comment by Rodrigonavarro23** [comment]
> Thanks! Always happy to support more data sources.
> 
> Do you know if the Dremio ODBC driver has some license the end user needs to accept, like the Oracle one? Because we might just add it to the Docker image if it doesn't.

Is open license, I could installing directly into the docker image

File: redash/query_runner/dremio.py (hunk 2 above)
Comment (kravets-levko) [reply redash/query_runner/dremio.py]: Isn't it better to specify driver name in env var?

**Comment by Rodrigonavarro23** [comment]
Hi, there is something else pending for this PR?

**Comment by guidopetri** [comment]
@Rodrigonavarro23 , thanks for the PR! We've updated a lot of things now that we're Community-driven so - if you're still interested in getting this merged - would you mind rebasing off master to re-run the CI, as well as updating merge conflicts?

We're trying to clean up our PR todo list, so if you're not interested, that's fine - we'll close the PR in about a week if we don't hear back. If you're interested in reopening the PR afterwards, we would also very much welcome that.

**Comment by justinclift** [comment]
Looking at the Dremio website, it looks like they've switched to a different ODBC driver in the meantime:

https://www.dremio.com/drivers/odbc/

```
Apache Arrow Flight SQL is a modern open source protocol, co-created by Dremio, for querying SQL-based
systems such as databases, data warehouses, and data lakehouses.
```

So the download in the Dockerfile would need to be updated to match.  The changes in the `requirements*.txt` files aren't needed any more either, as we already have `pyodbc` includes for other data sources. :smile:

**Pull request closed without merging**